app/src/main/java/com/onegini/mobile/exampleapp/view/activity/LoginActivity.java

```diff
@@ -145,13 +153,37 @@ private void onApplicationDetailsFetchFailed() {
   }
 
   @SuppressWarnings("unused")
+  @OnCheckedChanged(R.id.login_with_default_authenticator)
+  public void usePreferredAuthenticatorSwitchStateChanged() {
+    if (usePreferredAuthenticatorSwitchCompat.isChecked()) {
+      loginButton.setText(getString(R.string.btn_login_label));
+    } else {
+      loginButton.setText(getString(R.string.btn_login_with_label));
+    }
+  }
+
+  @SuppressWarnings("unused")
   @OnClick(R.id.login_button)
   public void loginButtonClicked() {
-    setProgressbarVisibility(true);
-
     final User user = listOfUsers.get(usersSpinner.getSelectedItemPosition());
-    loginUser(user.getUserProfile());
-    userIsLoggingIn = true;
+    if (usePreferredAuthenticatorSwitchCompat.isChecked()) {
+      setProgressbarVisibility(true);
```

Comment:
170-172 and 183-185 is duplicated, can be extracted to a separate method

Reply:
I don't think it's a good idea to extract just 2 lines into a new method:

`setProgressbarVisibility(true);
userIsLoggingIn = true;`
because `authenticateUser` methods aren't the same.